core/src/test/java/org/hibernate/ogm/backendtck/id/TableIdGeneratorInheritanceTest.java

```diff
@@ -0,0 +1,158 @@
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.backendtck.id;
+
+import java.util.UUID;
+import java.util.function.Consumer;
+import javax.persistence.Entity;
+import javax.persistence.EntityManager;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.Inheritance;
+import javax.persistence.Table;
+
+import org.hibernate.annotations.GenericGenerator;
+import org.hibernate.ogm.utils.jpa.OgmJpaTestCase;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static javax.persistence.InheritanceType.TABLE_PER_CLASS;
+import static org.fest.assertions.Assertions.assertThat;
+
+/**
+ * Test that @Inheritance(strategy = TABLE_PER_CLASS) with generated id work correct
+ */
+public class TableIdGeneratorInheritanceTest extends OgmJpaTestCase {
+
+	private static final String someTruckName = "someTruckName";
+	private static final String someOtherTruckName = "someOtherTruckName";
+	private EntityManager em;
+
+	@Before
+	public void setUp() {
+		em = getFactory().createEntityManager();
+	}
+
+	@After
+	public void tearDown() {
+		em.close();
+	}
+
+	@Test
+	public void testTableIdGeneratesAndEqualsSaved() {
+		Truck truck = new Truck();
+		doInTransaction( em -> em.persist( truck ) );
+
+		doInTransaction( em -> {
+			Truck loadedTruck = em.find( Truck.class, truck.getId() );
+			assertThat( loadedTruck.getId() ).isEqualTo( truck.getId() );
+		} );
+	}
+
+	@Test
+	public void testInheritedTableIdGenerates() {
+		Truck truck = new Truck();
+		doInTransaction( em -> em.persist( truck ) );
+
+		doInTransaction( em -> {
+			Truck loadedTruck = em.find( Truck.class, truck.getId() );
+			assertThat( loadedTruck.getId() ).isNotNull();
+		} );
+	}
+
+
+	@Test
+	public void testInheritedTableUpdates() {
+		Truck truck = new Truck();
+		truck.setName( someTruckName );
+
+		doInTransaction( em -> em.persist( truck ) );
+
+		doInTransaction( em -> {
+			Truck loadedTruck = em.find( Truck.class, truck.getId() );
+			assertThat( loadedTruck.getName() ).isEqualTo( someTruckName );
+
+			loadedTruck.setName( someOtherTruckName );
+			em.persist( loadedTruck );
+
+		} );
+
+		doInTransaction( em -> {
+			Truck loadedTruck = em.find( Truck.class, truck.getId() );
+			assertThat( loadedTruck.getName() ).isEqualTo( someOtherTruckName );
+		} );
+	}
+
+	private void doInTransaction(Consumer<EntityManager> action) {
+		em.getTransaction().begin();
+
+		action.accept( em );
+
+		em.getTransaction().commit();
+		em.clear();
+	}
+
+	@Test
+	public void testInheritedTableDelete() {
+		Truck truck = new Truck();
+		doInTransaction( em -> em.persist( truck ) );
+
+		doInTransaction( em -> {
+			Truck loadedTruck = em.find( Truck.class, truck.getId() );
+			em.remove( loadedTruck );
+		} );
+
+		doInTransaction( em -> {
+			Truck loadedTruck = em.find( Truck.class, truck.getId() );
+			assertThat( loadedTruck ).isNull();
```

Comment:
This test is not enough.
The fact that the entity is null doesn't mean that all the collections involved are empty. I think you can only check this with native queries.

Reply:
Do I understand correctly: should I check through a native request that the truck table is empty after removal?
Should I check that base table is empty and not exists?

Reply:
Mmmh .. let me check, maybe the mapping is not what I'm expecting

Reply:
Right, my bad. We don't create multiple tables in OGM for the same entity.
I will create a follow up issue to tihs one to add some test to check for the mapping in MongoDB for TABLE_PER_CLAS hierarchies, it seems that right now we only check for the single table per class case (SingleTableInheritanceTest)

Reply:
Is that enough of the test in a result?